Automattic/wp-calypso
# Server: Cleanup integration tests

#### Changes proposed in this Pull Request

This PR cleans up the server integration tests. Seems like we had a few tests that were unnecessary since #52947 - this PR removes them. Cleaning up also removes one of the few remaining instances of `use-sinon`.

#### Testing instructions

Verify integration tests pass: `yarn run test-integration client/server/api/integration/index.js`
